sources/sdk/k8s-operator/src/kube-interface.js

```diff
@@ -251,58 +251,37 @@ class KubeInterface {
     return response.unwrap(await endpoint.delete())
   }
 
-  async canI({ apiVersion, kind, namespace, verb }) {
-    const { apiGroup } = parseApiVersion(apiVersion)
-    const resp = await this.create({
-      apiVersion: 'authorization.k8s.io/v1',
-      kind: 'SelfSubjectAccessReview',
-      spec: {
-        resourceAttributes: {
-          group: apiGroup,
-          resource: kind.toLowerCase(),
-          verb,
-          namespace
+  async checkACL (rscKind) {
+    return async ({ apiVersion, kind, namespace, verb }) => {
+      const { apiGroup } = parseApiVersion(apiVersion)
+      const resp = await this.create({
+        apiVersion: 'authorization.k8s.io/v1',
+        kind: rscKind,
+        spec: {
+          resourceAttributes: {
+            group: apiGroup,
+            resource: kind.toLowerCase(),
+            verb,
+            namespace
+          }
         }
-      }
-    })
+      })
 
-    return resp.status.allowed
+      return resp.status
+    }
   }
 
-  async canThey({ apiVersion, kind, namespace, verb }) {
-    const { apiGroup } = parseApiVersion(apiVersion)
-    const resp = await this.create({
-      apiVersion: 'authorization.k8s.io/v1',
-      kind: namespace ? 'LocalSubjectAccessReview' : 'SubjectAccessReview',
-      spec: {
-        resourceAttributes: {
-          group: apiGroup,
-          resource: kind.toLowerCase(),
-          verb,
-          namespace
-        }
-      }
-    })
-
-    return resp.status.allowed
+  async canI(rsc) {
+    return (await this.checkACL('SelfSubjectAccessReview')(rsc)).allowed
   }
 
-  async myAccessRules({ apiVersion, kind, namespace, verb }) {
-    const { apiGroup } = parseApiVersion(apiVersion)
-    const resp = await this.create({
-      apiVersion: 'authorization.k8s.io/v1',
-      kind: 'SelfSubjectRulesReview',
-      spec: {
-        resourceAttributes: {
-          group: apiGroup,
-          resource: kind.toLowerCase(),
-          verb,
-          namespace
-        }
-      }
-    })
+  async canThey(rsc) {
```

Comment:
The API is unclear, what's the input ? It seems to be a Kubernetes resource, and yet on the next line you use `rsrc.namespace` instead of `rsrc.metadata.namespace`.